### PR TITLE
DevWs for Sinope Technologies containing containing VA4200WZ-VA4200ZB Sinope Valve

### PR DESCRIPTION
**Pull request by SmartThings**

### CODE REVIEW

File: devicetypes/sinope-technologies/va4200wz-va4200zb-sinope-valve.src/va4200wz-va4200zb-sinope-valve.groovy [hunk 1]
```diff
@@ -286,6 +284,58 @@ private int get_LOG_TRACE() {
 	return 5
 }
 
+/**
+   * Is valve should be false if the device is a sensor
+   */
+private def convertVoltToPercent(value) {
+    def NBDEPOINTS_DEG = 6;
+    def LevelValue;
```
Comment (KKlimczukS) [inline devicetypes/sinope-technologies/va4200wz-va4200zb-sinope-valve.src/va4200wz-va4200zb-sinope-valve.groovy:292]: Please start the variable name from a lower case.

File: devicetypes/sinope-technologies/va4200wz-va4200zb-sinope-valve.src/va4200wz-va4200zb-sinope-valve.groovy [hunk 2]
```diff
@@ -286,6 +284,58 @@ private int get_LOG_TRACE() {
...
+    // def LevelsTable = [0, 20000, 40000, 60000, 80000, 100000];
```
Comment (KKlimczukS) [inline devicetypes/sinope-technologies/va4200wz-va4200zb-sinope-valve.src/va4200wz-va4200zb-sinope-valve.groovy:293]: Is this block of commented lines necessary here ?

File: devicetypes/sinope-technologies/va4200wz-va4200zb-sinope-valve.src/va4200wz-va4200zb-sinope-valve.groovy [hunk 3]
```diff
@@ -286,6 +284,58 @@ private int get_LOG_TRACE() {
...
+    //   {volt: 1, percent: 0},
+    //   {volt: 2.28, percent: 20},
+    //   {volt: 2.46, percent: 40},
+    //   {volt: 2.64, percent: 60},
+    //   {volt: 2.82, percent: 80},
+    //   {volt: 3, percent: 100}
+    // ];
+
+    // if (isValve) {
+    def LevelsTable = [0, 20000, 40000, 60000, 80000, 100000];
```
Comment (KKlimczukS) [inline devicetypes/sinope-technologies/va4200wz-va4200zb-sinope-valve.src/va4200wz-va4200zb-sinope-valve.groovy:305]: Please start the variable name from a lower case.

File: devicetypes/sinope-technologies/va4200wz-va4200zb-sinope-valve.src/va4200wz-va4200zb-sinope-valve.groovy [hunk 4]
```diff
@@ -286,6 +284,58 @@ private int get_LOG_TRACE() {
...
+    def AnglesTable = [30, 55, 56, 57, 58.5, 60];
```
Comment (KKlimczukS) [inline devicetypes/sinope-technologies/va4200wz-va4200zb-sinope-valve.src/va4200wz-va4200zb-sinope-valve.groovy:306]: Please start the variable name from a lower case.

File: devicetypes/sinope-technologies/va4200wz-va4200zb-sinope-valve.src/va4200wz-va4200zb-sinope-valve.groovy [hunk 5]
```diff
@@ -286,6 +284,58 @@ private int get_LOG_TRACE() {
...
+
+    if (value > AnglesTable[NBDEPOINTS_DEG - 1]) { // if the value of the angle is greater than the maximum
+      value = AnglesTable[NBDEPOINTS_DEG - 1]; // use the maximum value instead
+    }
+
+    def index = 1;
+    for (index; value > AnglesTable[index]; index++) {
+    }
+
+    def RatioBetweenPointXandY = (LevelsTable[index] - LevelsTable[index - 1]) / (AnglesTable[index] - AnglesTable[index - 1]);
```
Comment (KKlimczukS) [inline devicetypes/sinope-technologies/va4200wz-va4200zb-sinope-valve.src/va4200wz-va4200zb-sinope-valve.groovy:325]: Please start the variable name from a lower case.

File: devicetypes/sinope-technologies/va4200wz-va4200zb-sinope-valve.src/va4200wz-va4200zb-sinope-valve.groovy [hunk 6]
```diff
@@ -286,6 +284,58 @@ private int get_LOG_TRACE() {
...
+    def AngleToAdd = LevelsTable[index] - (AnglesTable[index] * RatioBetweenPointXandY);
```
Comment (KKlimczukS) [inline devicetypes/sinope-technologies/va4200wz-va4200zb-sinope-valve.src/va4200wz-va4200zb-sinope-valve.groovy:326]: Please start the variable name from a lower case.

File: devicetypes/sinope-technologies/va4200wz-va4200zb-sinope-valve.src/va4200wz-va4200zb-sinope-valve.groovy [hunk 7]
```diff
@@ -286,6 +284,58 @@ private int get_LOG_TRACE() {
...
+    def LevelWithFactor = (RatioBetweenPointXandY * value) + AngleToAdd;
```
Comment (KKlimczukS) [inline devicetypes/sinope-technologies/va4200wz-va4200zb-sinope-valve.src/va4200wz-va4200zb-sinope-valve.groovy:327]: Please start the variable name from a lower case.

File: devicetypes/sinope-technologies/va4200wz-va4200zb-sinope-valve.src/va4200wz-va4200zb-sinope-valve.groovy [hunk 8]
```diff
@@ -69,9 +69,9 @@ def refresh() {
     def cmds = []
     cmds += zigbee.readAttribute(0x0006, 0x0000)//refresh on/off
     cmds += zigbee.readAttribute(0x0000, 0x0007)//refresh power source
-    cmds += zigbee.readAttribute(0x0001, 0x0021)//refresh battery percentage remaining
+    cmds += zigbee.readAttribute(0x0001, 0x0020)//refresh battery percentage remaining
```
Comment (KKlimczukS) [inline devicetypes/sinope-technologies/va4200wz-va4200zb-sinope-valve.src/va4200wz-va4200zb-sinope-valve.groovy:72]: Shouldn't You also change the text in this comment ? (percentage -> voltage)

File: devicetypes/sinope-technologies/va4200wz-va4200zb-sinope-valve.src/va4200wz-va4200zb-sinope-valve.groovy [hunk 9]
```diff
@@ -69,9 +69,9 @@ def refresh() {
...
     cmds += zigbee.configureReporting(0x0006, 0x0000, 0x10, 0, 600, null)//configure reporting on/off min: 0sec, max 600sec
-    cmds += zigbee.configureReporting(0x0001, 0x0021, 0x20, 60, 60*60, 1)//configure reporting battery percentage remaining min: 6sec, max 1hour
+    cmds += zigbee.configureReporting(0x0001, 0x0020, 0x20, 60, 60*60, 1)//configure reporting battery percentage remaining min: 6sec, max 1hour
```
Comment (KKlimczukS) [inline devicetypes/sinope-technologies/va4200wz-va4200zb-sinope-valve.src/va4200wz-va4200zb-sinope-valve.groovy:74]: Shouldn't You also change the text in this comment ? (percentage -> voltage)

**Review by KKlimczukS — CHANGES_REQUESTED** [review]

**Comment by k-morawski** [comment]
Dear @vbeauregard-Sinope, could you please check and implement @KKlimczukS review feedback? Many thanks!

**Comment by aguilbault-Sinope** [comment]
@KKlimczukS @k-morawski  
This request can be rejected, we have created a new pull request with the Smartthings Workspace #65711 with the change suggested by KKlimczukS

**Comment by KKlimczukS** [comment]
> @KKlimczukS @k-morawski
> This request can be rejected, we have created a new pull request with the Smartthings Workspace #65711 with the change suggested by KKlimczukS

@vbeauregard-Sinope 
Thank You. This pull request can be closed now.

**Comment by KKlimczukS** [comment]
According to the above comment, I'm closing this PR.

**Pull request closed without merging**